src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -504,8 +504,9 @@ class Address {
         if (size == 0) // It's a byte
           i->f(_ext.shift() >= 0, 12);
         else {
-          if (_ext.shift() > 0)
+          if (_ext.shift() > 0) {
             assert(_ext.shift() == (int)size, "bad shift");
+          }
```

Comment:
D'oh. So this is semantically the same as:

```
 assert(_ext.shift() <= 0 || _ext.shift() == (int)size, "bad shift");
```

...or, if we expect shift to be non-negative:

```
 assert(_ext.shift() == 0 || _ext.shift() == (int)size, "bad shift");
```

Reply:
Indeed. Are you suggesting to change it?  I usually try to avoid changing code style when doing a fix of such a nature.

Reply:
I feels odd to have empty method body when asserts are disabled. So yes, I suggest to subsume the check into the assert itself.

Reply:
Fair, I updated the PR. Thanks for the suggestion 🙂